spec/classes/profile/kubernetes/dns_server_spec.rb

```diff
@@ -69,6 +69,11 @@
           .with_content("::1 localhost ip6-localhost ip6-loopback\n")
       end
 
+      it do
+        is_expected.to contain_file('/etc/dnsmasq.d/smartconnect')
+          .with_content("server=/sc.default.invalid/192.0.2.7\n")
```

Comment:
I'd like to see another test for when the private zones aren't set, so that that behavior is mapped out. No need to like, expect it not to contain a file, but I'd at least like to be sure it compiles. An example of testing multiple cluster configs: https://github.com/mlibrary/nebula/blob/master/spec/classes/profile/kubernetes/kubeadm_spec.rb

Reply:
Thanks -- wasn't sure of the best place to test it. FWIW `second_cluster` does not have it configured in `spec/fixtures/hiera/kubernetes/default.yaml` - are you suggesting another test here in `dns_server_spec.rb`?